physics/MP/Morrison_Gettelman/aerinterp.F90

```diff
@@ -426,9 +813,157 @@ SUBROUTINE aerinterpol( me,master,nthrds,npts,IDATE,FHOUR,iflip, jindx1,jindx2,
 
       RETURN
       END SUBROUTINE aerinterpol
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!     private subroutines
+      subroutine fdnx_fname(lyear, lmn, ldy, fname)
+      integer, intent(inout) ::lyear, lmn, ldy
+      character, intent(out) :: fname*50
+      integer, dimension(12) :: mndy
+      character myr*4, mn*2, dy*2
+      data mndy/31, 28, 31,30,31,30,31,31,30,31,30,31/
+      ldy=ldy+1
+      if(lmn==12) then
+        if(ldy>mndy(12)) then
+          ldy=1
+          lmn=1
+          lyear=lyear+1
+        end if
+      else if(lmn==2) then
+        if (mod(lyear,4)==0) then
+          if(ldy>mndy(2)+1) then
+            ldy=1
+            lmn=3
+          end if
+        else
+          if(ldy>mndy(2)) then
+            ldy=1
+            lmn=3
+          end if
+        end if
+      else
+        if(ldy>mndy(lmn)) then
+          ldy=1
+          lmn=lmn+1
+        end if
+      end if
+       write(myr,'(i4.4)') lyear
+       write(mn,'(i2.2)') lmn
+       write(dy,'(i2.2)') ldy
+      fname="merra2_"//myr//mn//dy//".nc"
+      RETURN
+      END SUBROUTINE fdnx_fname
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      subroutine read_netfaer_dl(fname, nf, iflip,nt, errmsg,errflg)
+      use machine, only: kind_phys, kind_io4
+      use aerclm_def
+      use netcdf
+      integer, intent(in) :: iflip, nf, nt
+      character,intent(in)   :: fname*50
+      integer, intent(inout) :: errflg
```

Comment:
Same question about intents for errflg and errmsg

Reply:
same comments as above.

Reply:
change intent(inout) to intent(out) for all related errmsg and errflag and committed